### PR TITLE
Preference Service Tests

### PR DESCRIPTION
**Pull request by 0xDaedalus**
This PR lays the groundwork for testing codepaths that rely on indexedDB - and adds some basic preferenceService tests to serve as an example of how we may want to test a service going forward.

### CODE REVIEW

**Review by greg-nagy — APPROVED** [review]
CleanShot 2022-07-14 at 16 30 25

that's all I can say ... other than spectacular 🚀

File: background/services/name/resolvers/address-book.ts [hunk 1]
```diff
@@ -17,12 +18,20 @@ export default function addressBookResolverFor(
...
     },
 
-    lookUpAddressForName:
-      preferenceService.lookUpAddressForName.bind(preferenceService),
+    async lookUpAddressForName(nameOnNetwork: NameOnNetwork) {
+      return preferenceService.lookUpAddressForName.call(
+        preferenceService,
+        nameOnNetwork
+      )
+    },
```
Comment (Shadowfiend) [inline background/services/name/resolvers/address-book.ts:26]: If you're making this a full on method you don't need to use `call`. `preferenceService.lookUpAddressForName(nameOnNetwork)` will bind `this` without extra ceremony.

File: background/services/preferences/index.ts [hunk 2]
```diff
@@ -102,19 +102,19 @@ export default class PreferenceService extends BaseService<Events> {
     this.emitter.emit("addressBookEntryModified", newEntry)
   }
 
-  async lookUpAddressForName({
+  lookUpAddressForName({
     name,
     network,
-  }: NameOnNetwork): Promise<AddressOnNetwork | undefined> {
+  }: NameOnNetwork): AddressOnNetwork | undefined {
```
Comment (Shadowfiend) [inline background/services/preferences/index.ts:108]: Thisss was written as a promise with the expectation that the address book was going to move to IndexedDB fwiw.

Comment (Shadowfiend) [reply background/services/preferences/index.ts:108]: (I realize I did not say that, or write it, and therefore there was no reason to know it lol. Just clarifying.)

Comment (greg-nagy) [reply background/services/preferences/index.ts:108]: This is part of the Indiana Jones vibes :)

**Pull request merged**